ddtrace/tracer/tracer.go

```diff
@@ -323,27 +323,40 @@ func (t *tracer) worker(tick <-chan time.Time) {
 // finishedTrace holds information about a trace that has finished, including its spans.
 type finishedTrace struct {
 	spans    []*span
-	decision samplingDecision
+	willSend bool // willSend indicates whether the trace will be sent to the agent.
 }
 
 // sampleFinishedTrace applies single-span sampling to the provided trace, which is considered to be finished.
 func (t *tracer) sampleFinishedTrace(info *finishedTrace) {
-	if info.decision == decisionKeep {
+	// The trace is kept, no need to run single span sampling rules.
+	// It is important that to check the sampling decision before checking the
```

Comment:
```suggestion
	// It is important to check the sampling decision before checking the
```
Can we also add a "why" for this comment? (I assume the reason is so we don't try to single span sample a trace that is already being sent to the backend since the single span tags should only be sent when doing the single span sampling? Which itself is important so users can see _why_ spans/traces were kept)

Reply:
if a trace is kept, we don't care if there are rules, and we don't need to update metrics like droppedP0Traces. 
If a trace is dropped, 
- if the rules are present, run sampling, update metrics according to the results
- if not present, don't run sampling, update metrics (`droppedP0Traces +=1, droppedP0Spans += len(spans)`)

Reply:
I chose to remove that comment. We shouldn't touch the sampling if the trace if kept, so to avoid the confusion, I removed it